sql/hive/src/main/scala/org/apache/spark/sql/hive/hiveWriterContainers.scala

```diff
@@ -211,18 +211,17 @@ private[spark] class SparkHiveDynamicPartitionWriterContainer(
       }
     }
 
-    val dynamicPartPath = dynamicPartColNames
-      .zip(row.toSeq.takeRight(dynamicPartColNames.length))
-      .map { case (col, rawVal) =>
-        val string = if (rawVal == null) null else convertToHiveRawString(col, rawVal)
-        val colString =
-          if (string == null || string.isEmpty) {
-            defaultPartName
-          } else {
-            FileUtils.escapePathName(string, defaultPartName)
-          }
-        s"/$col=$colString"
-      }.mkString
+    val dynamicPartPath = dynamicPartColNames.zipWithIndex.map { case (colName, i) =>
+      val rawVal = row.get(i, schema(colName).dataType)
```

Comment:
Is this the right way to get the schema of dynamic partition column? I'm not sure what's the relation between `schema` and this `row` and `dynamicPartColNames`.
cc @marmbrus